torch_xla/csrc/ir.h

```diff
@@ -235,7 +247,8 @@ class Node {
   std::vector<NodePtr> operands_;
   // Outputs do not hold references on the nodes, and neither do the uses, since
   // otherwise we get into circular reference counting.
-  std::vector<Output> operands_as_outputs_;
+  std::vector<torch::lazy::Output> operands_as_outputs_;
+  std::vector<Output> operands_as_outputs_with_shape_;
```

Comment:
WE shouldn't introduce a new vector just because our `Output` is different than `torch::lazy::Output` (I think the only difference is our version has `xla::shape()`).

@wconstab I remembered we had a conversation regarding this, what's the reason you removed `shape`  within the `Output`? I remembered you mentioned that it was not being used anywhere or something.

Reply:
I don't mind adding shape back to output, although, I also want to discuss Output vs Value with you guys and see if we can merge the two.

I also intended to put shape() api back on Node base class.  I'm surprised you didn't remind me, i thought you needed it.  If it helps, I can do it soon.

Reply:
Ah OK. XLANode needs to have `xla::shape`, our hope is that we can use `lazy::shape` in the `IR Node and XLATensor` layer and `xla::shape` from lowering part and below. Since we hasn't start using codegen, no `lazy::shape` is being provided(from formula or meta tensor) so we hasn't run into this issue yet.

The long term plan is still to use `lazy::shape` in the non-lowering layer but until all of the op we currently supported today has a meta kernel or formula, we can't do this switch. This will become a larger issue when we do the phase 3 migration since LazyTensorBase class will always use `lazy::shape`.

Reply:
OK, well as long as it is generally helpful / not harmful for you, I will move shape to Node base soon.